src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -505,3 +536,18 @@ export function parseKeyValueAssignment(message: string): { key: string; value:
 		value: deserialized.substring(1 + separatorIndex)
 	};
 }
+
+
+export function parseMarkSequence(sequence: string[]): { id?: string; hidden?: boolean } {
+	let id = undefined;
+	let hidden = false;
+	for (const property of sequence) {
+		if (property === 'true') {
```

Comment:
Shouldn't this be matching `'Hidden'` instead? That's what this was supposed to mean:

```
OSC 633 ; SetMark [; Id=<string>] [; Hidden]
```

That way we could always tack on extra properties that may be booleans as well:

```
OSC 633 ; SetMark [; Id=<string>] [; Hidden] [; IsShiny]
```